### PR TITLE
Add top level error logging.

### PR DESCRIPTION
**Pull request by Xjph**

### CODE REVIEW

File: ObservatoryCore/ObservatoryCore.cs [hunk 1]
```diff
@@ -10,13 +10,39 @@ class ObservatoryCore
...
-                Properties.Core.Default.CoreVersion = version;
-                Properties.Core.Default.Save();
+                if (Properties.Core.Default.CoreVersion != version)
+                {
+                    Properties.Core.Default.Upgrade();
+                    Properties.Core.Default.CoreVersion = version;
+                    Properties.Core.Default.Save();
+                }
+                BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
+                throw new Exception("Test Exception", new Exception("Inner Exception Test"));
```
Comment (fredjk-gh) [inline ObservatoryCore/ObservatoryCore.cs:22]: Should this still be here?

File: ObservatoryCore/ObservatoryCore.cs [hunk 2]
```diff
@@ -10,13 +10,39 @@ class ObservatoryCore
...
             }
-            BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
+            catch (Exception ex)
+            {
+                var docPath = System.Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments);
+                var errorMessage = new System.Text.StringBuilder();
+                errorMessage.AppendLine($"Error encountered in Elite Observatory {version}.");
```
Comment (fredjk-gh) [inline ObservatoryCore/ObservatoryCore.cs:28]: Optional: Consider chaining:

errorMessage
    .AppendLine(...)
    .AppendLine(...)
    .AppendLine();

(Of course, however the editor formats it. Below as well.)

**Review by fredjk-gh — CHANGES_REQUESTED** [review]
Only requesting changes because the test exception is left in there. I assume we don't want that committed.

File: ObservatoryCore/ObservatoryCore.cs (hunk 1 above)
Comment (Xjph) [reply ObservatoryCore/ObservatoryCore.cs:22]: Definitely shouldn't!

**Review by fredjk-gh — APPROVED** [review]

**Pull request merged**